File: ttms-backend/internal/model/request/plan.go
/**
 * @Author: lenovo
 * @Description:
 * @File:  plan
 * @Version: 1.0.0
 * @Date: 2023/06/04 4:37
 */

package request

type CreatePlan struct {
	MovieID   uint    `json:"movieID" binding:"required"`
	CinemaID  uint    `json:"cinemaID" binding:"required"`
	Price     float64 `json:"price" binding:"required"`
	StartTime int64   `json:"startTime" binding:"required"`
}

type DelPlan struct {
	PlanID uint `json:"planID" binding:"required"`
}

File: ttms-front/src/apis/plan.js
import request from "./base";
export const getPlan = (page) => {
  return request({
    url: `/plan/list/${page}`,
    method: "GET",
    data: page,
  });
};
export const deletePlan = (planID) => {
  return request({
    url: "/plan/delete",
    method: "DELETE",
    data: { planID },
  });
};
export const createPlan = (movieID, cinemaID, price, startTime) => {
  return request({
    url: "/plan/create",
    method: "POST",
    data: {
      movieID,
      cinemaID,
      price,
      startTime,
    },
  });
};
